src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSpark.java

```diff
@@ -31,7 +32,9 @@
     private static final long serialVersionUID = 1L;
     @VisibleForTesting static final Long MAX_KMER_FREQ = 3L;
     private static final int REF_RECORD_LEN = 10000;
-    private static final int CHUNK_SIZE = 3000000;
+    // to give a task size of 100Kb -- the max size recommended in the Spark documentation
+    private static final int REF_RECORDS_PER_PARTITION = 102400 / REF_RECORD_LEN;
```

Comment:
The precise message is "Stage 0 contains a task of very large size (100 KB). The maximum recommended task size is 100 KB."  This is using the code you're examining with the task size calculated to be 100Kb.  If I revise the code to use some other task size, this number tracks it.  I think it really is just the partition size.
The issue is that the pipeline is going to transform inputs to outputs, and the outputs can be any size at all relative to the inputs (the "fanout" being a number probably known only to the application designer, and not easily inferrable by the framework), and all the outputs from all the cores have to fit in available executor memory (after deducting storage and overhead).  This is a crazy complex calculation, made even more difficult by the plethora of config params.
I know I have a 64x inflation factor (one byte of input produces ~64 bytes of output), so I need to keep the task size down, figure out a way to instruct my users how large to make executors, or respond to the available memory by adjusting the task size.
This small a partition size is probably not optimal, but will almost certainly allow the program to run no matter how much memory executors are configured with and no matter how many cores each executor uses.  So I'm giving up.  The whole thing makes my brain hurt.